protocol/src/main/java/org/opensearch/sql/protocol/response/format/FlatResponseBase.java

```diff
@@ -84,6 +84,10 @@ protected List<List<String>> formatData(List<List<String>> lines) {
 
   protected String quoteIfRequired(String separator, String cell) {
     final String quote = "\"";
-    return cell.contains(separator) ? quote + cell.replaceAll("\"", "\"\"") + quote : cell;
+    if (cell != null && (cell.contains(separator) || cell.contains(quote))) {
+      return quote + cell.replaceAll(quote, quote + quote) + quote;
```

Comment:
No matter cell contains separator or quote, we always replace quote?

Reply:
If you read the RFC for the CSV output format, https://www.ietf.org/rfc/rfc4180.txt section 2.5 states:

> Each field may or may not be enclosed in double quotes (however some programs, such as Microsoft Excel, do not use double quotes  at all).  If fields are not enclosed with double quotes, then double quotes may not appear inside the fields.

So my understanding is that we would quote if the field contains separator or quote.